site/cds_rdm/ldap/utils.py

```diff
@@ -0,0 +1,119 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# CDS-RDM is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+
+"""CDS-RDM ldap serializers."""
+
+from functools import partial
+
+from cds_rdm.ldap.errors import InvalidLdapUser
+from invenio_accounts.models import User
+from invenio_oauthclient.models import UserIdentity
+from invenio_userprofiles import UserProfile
+
+
+def serialize_ldap_user(ldap_user_data, log_func=None):
+    """Create ldap user."""
+
+    def serialize(ldap_user_data):
+        decoded_data = {}
+        for key, value in ldap_user_data.items():
+            # get first value of each field from LDAP
+            # decode bytes to UTF-8
+            decoded_data[key] = value[0].decode("utf8")
+        serialized_data = dict(
+            user_email=decoded_data["mail"].lower(),
+            user_profile_full_name=decoded_data["displayName"],
+            user_identity_id=decoded_data["uidNumber"],
+            cern_account_type=decoded_data["cernAccountType"],
+            remote_account_department=decoded_data["department"],
+            keycloak_id=decoded_data.get("name", decoded_data["cn"]),
```

Comment:
I am not 100% sure about this, maybe we should rely only on one, the `cn` or `name`, prior agreement wit the Auth team to make sure which one will map to the keycloak id all the time